flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -821,7 +821,7 @@ class TaskManager(
     val logFilePathOption = Option(config.configuration.getString(
       ConfigConstants.TASK_MANAGER_LOG_PATH_KEY, System.getProperty("log.file")));
     logFilePathOption match {
-      case None => throw new IOException("TaskManager log files are unavailable. " +
+      case None => sender ! new IOException("TaskManager log files are unavailable. " +
```

Comment:
In order to send an `Exception` over the wire to the sender it has to be packed into `akka.actor.Status.Failure`. Otherwise it will only be sent as a normal response type.